### ISSUE TITLE
write_zip() doesn't store permissions into ZIP archives 

### ISSUE THREAD

**Issue by mashabow** [PyFilesystem/pyfilesystem2#60]
```console
$ ipython
Python 3.6.1 (default, Jul 19 2017, 13:24:40)
Type 'copyright', 'credits' or 'license' for more information
IPython 6.1.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from fs.zipfs import ZipFS

In [2]: with ZipFS('test.zip', write='True') as z:
   ...:     z.makedir('foo', 0o777)
   ...:     z.makedir('bar', 0o755)
   ...:     z.makedir('baz', 0o700)
   ...:

In [3]:
Do you really want to exit ([y]/n)? y

$ zipinfo test.zip
Archive:  test.zip   274 bytes   3 files
?rw-------  2.0 unx        0 b- stor 26-Jul-17 15:00 bar/
?rw-------  2.0 unx        0 b- stor 26-Jul-17 15:00 baz/
?rw-------  2.0 unx        0 b- stor 26-Jul-17 15:00 foo/
3 files, 0 bytes uncompressed, 0 bytes compressed:  0.0%
```

It seems we have to set `ZipInfo.external_attr` in [`write_zip()`](https://github.com/PyFilesystem/pyfilesystem2/blob/44260e742b42b0e124e9f87dd115a70c4c857fc7/fs/compress.py#L25).

**Comment by willmcgugan** [comment]
To be honest, I didn't realise that Zip files stored permissions info. But since they do, I agree ZipFS should write it.

**Comment by willmcgugan** [comment]
Should be fixed in 2.0.15 Thanks to @althonos
